.github/workflows/sanitizers.yml

```
@@ -0,0 +1,54 @@
+name: Sanitizers
+
+on: [pull_request, push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - sanitizer-name: asan with sleipnir
+            cmake-args: "-DCMAKE_BUILD_TYPE=Asan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: tsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan with sleipnir
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: asan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Asan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: tsan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+
+    name: "${{ matrix.sanitizer-name }}"
```

You can make sanitizer-name just be the name of the sanitizer and append the two variables like this.
```suggestion
    name: "${{ matrix.sanitizer-name }} with ${{ matrix.optimizer_backend }}"
```